### ISSUE TITLE
Fastlane fails during gym about the ditto command

### ISSUE THREAD

**Issue by pRizz** [fastlane/fastlane#1460]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
Fastlane fails in Jenkins during gym about the ditto command. Fastlane on my local computer works fine, but this is not a practical solution because the build takes ~ an hour. Archiving manually in Xcode works fine. 

##### Complete output when running fastlane, including the stack trace and command used
The entire logfile is ~200MB with sensitive information, so I will not post it.
This is the failing line of output.

```
Ditto /Users/iosjenkins/Library/Developer/Xcode/DerivedData/MyApp-dikjbbwmnvizjmdpbkrxixczfmqo/Build/Intermediates/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Development-iphoneos/MyApp.build/DerivedSources/MyApp-Swift.h /Users/iosjenkins/Library/Developer/Xcode/DerivedData/MyApp-dikjbbwmnvizjmdpbkrxixczfmqo/Build/Intermediates/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Development-iphoneos/MyApp.build/Objects-normal/armv7/MyApp-Swift.h
    cd /Users/iosjenkins/.jenkins/jobs/MyAppDevelopmentNightly/workspace/MyApp
    export PATH="/Applications/Xcode7.2.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode7.2.1.app/Contents/Developer/usr/bin:/usr/local/bin/:/usr/bin:/bin:/usr/sbin:/sbin"
    /usr/bin/ditto -rsrc /Users/iosjenkins/Library/Developer/Xcode/DerivedData/MyApp-dikjbbwmnvizjmdpbkrxixczfmqo/Build/Intermediates/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Development-iphoneos/MyApp.build/Objects-normal/armv7/MyApp-Swift.h /Users/iosjenkins/Library/Developer/Xcode/DerivedData/MyApp-dikjbbwmnvizjmdpbkrxixczfmqo/Build/Intermediates/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Development-iphoneos/MyApp.build/DerivedSources/MyApp-Swift.h
ditto: can't get real path for source '/Users/iosjenkins/Library/Developer/Xcode/DerivedData/MyApp-dikjbbwmnvizjmdpbkrxixczfmqo/Build/Intermediates/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/MyApp.build/Development-iphoneos/MyApp.build/Objects-normal/armv7/MyApp-Swift.h'
Command /usr/bin/ditto failed with exit code 1
```

Here is the variable dump:
```
[15:11:17]: Exit status: 65
[15:11:17]: Variable Dump:
[15:11:17]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios pushDevBuild", :SIGH_PROFILE_PATH=>"/Users/iosjenkins/.jenkins/jobs/MyAppDevelopmentNightly/workspace/AppStore_com.abc.MyApp.debug.ShareExtension.mobileprovision", :SIGH_PROFILE_PATHS=>["/Users/iosjenkins/.jenkins/jobs/MyAppDevelopmentNightly/workspace/AppStore_com.abc.MyApp.debug.mobileprovision", "/Users/iosjenkins/.jenkins/jobs/MyAppDevelopmentNightly/workspace/AppStore_com.abc.MyApp..debug.ShareExtension.mobileprovision"], :SIGH_UDID=>"abce227-3c14-1234-8920-b76e7912302e", :SIGH_PROFILE_TYPE=>"app-store"}
```

### Configuration Files

My fastlane script performs sigh, gym, get_version_number, get_build_number, commit_version_bump, add_git_tag, pilot, git_pull, push_to_git_remote.

### Environment

fastlane version (fastlane -v): fastlane 1.63.1
Do you use bundler, rbenv or rvm? No

### More
My project is very large ~ 1000 swift files. Each line of output the xcodebuild compilation is 110k characters of text.

**Comment by pRizz** [comment]
Some forums, like SO, mentioned to clear Xcode's DerivedData folder, but this did not fix it.

**Comment by pRizz** [comment]
Now, local archive fails, without fastlane at all. The problem seems to have escalated to normal archiving.

**Comment by pRizz** [comment]
This was an issue with the swiftc compiler, and is fixed in Xcode 7.3 beta: [https://bugs.swift.org/browse/SR-280](https://bugs.swift.org/browse/SR-280) 
It is strange that fastlane started getting affected a few days before archiving manually.

Since Apple needs builds from a non-beta Xcode, I implemented the workaround, here: [https://github.com/tgebarowski/swiftc-wrapper](https://github.com/tgebarowski/swiftc-wrapper)

**Comment by vpolouchkine** [comment]
It looks like this has been fixed by Apple at this point, so I’m closing it. If this is still a problem, feel free to re-open a new issue and I’d be happy to help :+1:
